src/shared/containers/ResourceViewContainer.tsx

```diff
@@ -726,10 +774,53 @@ const ResourceViewContainer: React.FunctionComponent<{
                       <Alert
                         type="error"
                         message={
-                          <>
-                            <DeleteOutlined /> This resource is deprecated. You
-                            cannot modify it.
-                          </>
+                          <div>
+                            <DeleteOutlined /> This resource is deprecated and
+                            not modifiable.
+                            {// Don't show the undo deprecated button if the resource is
+                            // of any unsupported resource. However, it needs to be shown
+                            // e.g. for custom types of resources.
+                            !resource['@type']?.includes(
+                              'View' || 'Resolver' || 'Storage' || 'Schema'
+                            ) ? (
+                              <>
+                                <br />
+                                {// If not newest revision, then don't show the button
+                                resource._rev === latestResource._rev ? (
+                                  <Button
+                                    icon={<UndoOutlined />}
+                                    style={{
+                                      marginTop: '10px',
+                                      marginBottom: '5px',
+                                    }}
+                                    onClick={async () => {
+                                      unDeprecateResource();
+                                    }}
+                                  >
```

Comment:
no need for async here, just pass the fn to on click
`onClick={() => unDeprecateResource()}`

Reply:
Good catch! Fixed it.